app/controllers/Application.scala

```diff
@@ -138,7 +138,7 @@ class Application @Inject() (
 
   def onHandlerNotFound(route: String) = Action { implicit request =>
     if (route.endsWith("/")) {
-      MovedPermanently(request.path.take(request.path.length - 1))
+      MovedPermanently("//" + request.host + "/" + request.path.take(request.path.length - 1).dropWhile(_ == '/'))
```

Comment:
Thanks a lot for reporting this and working on a fix.

`request.host` is coming from the `Host` header. I don't think the `Host` header spoofable in browsers, except by setting DNS to point at playframework.com, which doesn't matter. What do you think? We might not need to do the `"//" + request.host + "/"` part anyway because it looks like `.dropWhile(_ == '/')` drops leading slashes.

Another idea is to use a regex to restrict the paths we allow redirects for, e.g. paths with at least one simple alphanumeric/underscore character between slashes, no escaping, no dots, etc: `/(\w+/)+`.